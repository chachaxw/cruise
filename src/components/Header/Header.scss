@import '../../styles/_variables';

.cruise-header {
    position: fixed;
    top: 0;
    z-index: 99;
    width: 100%;
    background-color: #FFFFFF;
    box-shadow: 1px 4px 10px rgba(0, 0, 0, .1);

    .header-wrapper {
        position: relative;
        max-width: 1200px;
        margin: 0 auto;
        display: flex;
        justify-content: flex-end;
    }

    .logo {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }

    .logo img {
        height: 40px;
        width: auto;
    }
}

.user-menu-area {
    position: relative;
    padding: 12px;
    display: inline-flex;
    align-items: center;

    &:hover .user-menu-list,
    .user-menu-list:hover {
        display: block;
    }
}

.user-avatar {
    .avatar-img {
        display: inline-block;
        width: 40px;
        height: 40px;
        margin-right: 10px;
        border-radius: 50%;
        vertical-align: middle;
        background: center no-repeat;
        background-color: #DDDDDD;
        background-size: cover;
        /* background-image: url(); */
    }

    .iconfont {
        color: #435466;
    }
}

.user-menu-list {
    position: absolute;
    right: 0;
    top: 100%;
    display: none;
    padding: 10px 0;
    background-color: #FFFFFF;
    box-shadow: 1px 4px 10px rgba(0, 0, 0, .1);

    .user-menu-item {
        padding: 8px 12px;
        color: #2D4054;
        display: flex;
        align-items: center;
        min-width: 128px;
        cursor: pointer;

        &:hover {
            background-color: #EFEFEF;
        }

        .menu-label {
            padding-left: 10px;
        }
    }
}