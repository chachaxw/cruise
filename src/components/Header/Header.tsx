import * as React from 'react';
import { Link } from 'react-router-dom';
import Wrapper from '../Wrapper/Wrapper';
import './Header.scss';
import logo from '../../logo.svg';

type MenuItem = {
    icon: string,
    label: string,
}

export default class Header extends React.Component {
    private menu: MenuItem[] = [
        { icon: 'icon-id-card', label: 'Profile' },
        { icon: 'icon-sign-in', label: 'Sign Out' },
    ];

    public renderUserMenuList() {
        return (
            <div className="user-menu-area">
                <div className="user-avatar">
                    <span className="avatar-img"></span>
                    <span className="iconfont icon-angle-down"></span>
                </div>
                <ul className="user-menu-list">
                    {this.menu && this.menu.length && this.menu.map((item, index) => (
                        <li className="user-menu-item" key={index}>
                            <span className={`iconfont ${item.icon}`}></span>
                            <span className="menu-label">{item.label}</span>
                        </li>
                    ))}
                </ul>
            </div>
        );
    }

    public render() {
        return (
            <header className="cruise-header">
                <Wrapper className="header-wrapper">
                    <Link to="/" className="logo">
                        <img src={logo}></img>
                    </Link>
                    {this.renderUserMenuList()}
                </Wrapper>
            </header>
        );
    }
}
