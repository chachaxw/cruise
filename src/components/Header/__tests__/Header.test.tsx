import * as React from 'react';
import { StaticRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Header from '../Header';

it('renders with correctly header content', () => {
    const testRenderer = TestRenderer.create(
        <StaticRouter context={{}}>
            <Header></Header>
        </StaticRouter>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
});