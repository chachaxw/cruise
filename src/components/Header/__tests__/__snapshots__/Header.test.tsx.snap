// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correctly header content 1`] = `
<header
  className="cruise-header"
>
  <div
    className="header-wrapper"
    style={Object {}}
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      <img
        src="logo.svg"
      />
    </a>
    <div
      className="user-menu-area"
    >
      <div
        className="user-avatar"
      >
        <span
          className="avatar-img"
        />
        <span
          className="iconfont icon-angle-down"
        />
      </div>
      <ul
        className="user-menu-list"
      >
        <li
          className="user-menu-item"
        >
          <span
            className="iconfont icon-id-card"
          />
          <span
            className="menu-label"
          >
            Profile
          </span>
        </li>
        <li
          className="user-menu-item"
        >
          <span
            className="iconfont icon-sign-in"
          />
          <span
            className="menu-label"
          >
            Sign Out
          </span>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
